src/NAppUpdate.Framework/Utils/PermissionsCheck.cs

```diff
@@ -22,9 +24,8 @@ public static bool HaveWritePermissionsForFolder(string path)
 			return HaveWritePermissionsForFileOrFolder(folder);
 		}
 
-		public static bool HaveWritePermissionsForFileOrFolder(string path)
-		{
-			var rules = Directory.GetAccessControl(path).GetAccessRules(true, true, typeof(SecurityIdentifier));
+		public static bool HaveWritePermissionsForFileOrFolder(string path) {
+			var rules = new FileSecurity(path, AccessControlSections.All).GetAccessRules(true, true, typeof(SecurityIdentifier));
```

Comment:
This functionality currently isn't covered by tests (I believe?), happy to fill this gap if needed.